src/components/Channel/Channel.js

```diff
@@ -1,148 +1,674 @@
-     * */
-    Attachment: PropTypes.oneOfType([PropTypes.node, PropTypes.elementType]),
-    /**
-     * If true, KeyboardCompatibleView wrapper is disabled.
-     *
-     * Channel component internally uses [KeyboardCompatibleView](https://github.com/GetStream/stream-chat-react-native/blob/master/src/components/KeyboardCompatibleView.js) component
-     * internally to adjust the height of Channel component when keyboard is opened or dismissed. This prop gives you ability to disable this functionality, in case if you
-     * want to use [KeyboardAvoidingView](https://facebook.github.io/react-native/docs/keyboardavoidingview) or you want to handle keyboard dismissal yourself.
-     * KeyboardAvoidingView works well when your component occupies 100% of screen height, otherwise it may raise some issues.
-     * */
-    disableKeyboardCompatibleView: PropTypes.bool,
-    /**
-     * Custom wrapper component that handles height adjustment of Channel component when keyboard is opened or dismissed.
-     * Defaults to [KeyboardCompatibleView](https://github.com/GetStream/stream-chat-react-native/blob/master/src/components/KeyboardCompatibleView.js)
-     *
-     * This prop can be used to configure default KeyboardCompatibleView component.
-     * e.g.,
-     * <Channel
-     *  channel={channel}
-     *  ...
-     *  KeyboardCompatibleView={(props) => {
-     *    return (
-     *      <KeyboardCompatibleView keyboardDismissAnimationDuration={200} keyboardOpenAnimationDuration={200}>
-     *        {props.children}
-     *      </KeyboardCompatibleView>
-     *    )
-     *  }}
-     * />
-     */
-    KeyboardCompatibleView: PropTypes.oneOfType([
-      PropTypes.node,
-      PropTypes.elementType,
-    ]),
-    /**
-     * Override mark channel read request (Advanced usage only)
-     *
-     * @param channel Channel object
-     * */
-    doMarkReadRequest: PropTypes.func,
-    /**
-     * Override send message request (Advanced usage only)
-     *
-     * @param channelId
-     * @param messageData Message object
-     * */
-    doSendMessageRequest: PropTypes.func,
-    /**
-     * Override update message request (Advanced usage only)
-     * @param channelId
-     * @param updatedMessage UpdatedMessage object
-     * */
-    doUpdateMessageRequest: PropTypes.func,
-    /** Disables the channel UI if channel is frozen */
-    disableIfFrozenChannel: PropTypes.bool,
-  };
-
-  static defaultProps = {
-    disableKeyboardCompatibleView: false,
-    disableIfFrozenChannel: true,
-    KeyboardCompatibleView,
-  };
-
-  render() {
-    const { t } = this.props;
-    if (!this.props.channel.cid) {
-      return <Text>{t('Please select a channel first')}</Text>;
+const Channel = (props) => {
+  const {
+    channel,
+    children,
+    disableIfFrozenChannel = true,
+    disableKeyboardCompatibleView = false,
+    emojiData = emojiDataDefault,
+    KeyboardCompatibleView = KeyboardCompatibleViewDefault,
+  } = props;
```

Comment:
the only ones I didn't destructure there are the loading indicators, and since we set their default values to our own components if the user doesn't create their own, i was thinking if there might be a small memory benefit to keeping them undefined until/unless we have to use them